artiq/compiler/transforms/llvm_ir_generator.py

```diff
@@ -1407,13 +1408,17 @@ def _quote_attributes():
                                          not types.is_c_function(typ.attributes[attr]))
                     if is_class_function:
                         attrvalue = self.embedding_map.specialize_function(typ.instance, attrvalue)
+                    if all_attrs_constant:
+                        inst = typ.instance if types.is_constructor(typ) else typ
```

Comment:
It is. E.g. in `class Foo:`, TConstructor is the type of the expression `Foo` whereas TInstance is the type of the expression `Foo()`. I'll merge this manually while fixing this.